src/features/unified-share-modal/EmailForm.js

```diff
@@ -164,32 +231,41 @@ class EmailForm extends React.Component<Props, State> {
         event.preventDefault();
 
         const { onSubmit, selectedContacts } = this.props;
-        const { message, contactsFieldError } = this.state;
+        const { message, contactsFieldError, selectedJustificationReason } = this.state;
 
         if (contactsFieldError !== '') {
             // Block submission if there's a validation error
             return;
         }
 
         const contactsError = this.validateContacts(selectedContacts);
-        if (contactsError) {
+        const justificationReasonError = this.validateJustificationReason();
+
+        if (contactsError || justificationReasonError) {
             return;
         }
 
         const emails = [];
         const groupIDs = [];
+        const restrictedExternalEmails = [];
 
         selectedContacts.forEach(({ type, value }) => {
             if (type === 'group') {
                 groupIDs.push(value);
             } else {
+                if (this.isRestrictedExternalEmail(value)) {
+                    restrictedExternalEmails.push(value);
+                }
                 emails.push(value);
             }
         });
+
         onSubmit({
             emails,
             groupIDs,
+            justificationReason: selectedJustificationReason,
```

Comment:
Alternatively we could avoid including these new value when they do not apply (currently they will be either `null` or an empty array, respectively), but I think this is a bit cleaner. Note that `UnifiedShareForm` processes only the data that it needs.